aip/client-libraries/4222.md

```diff
@@ -41,21 +142,22 @@ md := metadata.Pairs("x-goog-request-params",
   url.QueryEscape("parent") + "=" + url.QueryEscape(req.GetParent()))
 ```
 
-Much like URL parameters, if there is more than one key-value pair, the `&`
-character is used as the separator. At runtime, if a named parameter is unset
-on the request message, it **should not** be included in the headers.
+At runtime, if a field with the same name as the named parameter is unset on the
+request message, the key-value pair corresponding to that parameter **must not**
+be included in the routing header. If none of the parameters must be included in
+the routing header, the routing header **must not** be sent.
 
 If the [`google.api.http`][http] annotation contains `additional_bindings`,
-these patterns **should** be parsed for additional request parameters. Fields
-not duplicated in the top-level (or `additional_bindings`) pattern **should**
+these patterns **must** be parsed for additional request parameters. Fields
+not duplicated in the top-level (or `additional_bindings`) pattern **must**
 be included in request parameters, encoded in the same way.
 
-**Note:** There is no additional annotation used for this; the presence of the
-key in any of the standard fields (`get`, `post`, `put`, `patch`, `delete`) on
-the [`google.api.http`][http] annotation is sufficient.
+Much like URL parameters, if there is more than one key-value pair, the `&`
+character is used as the separator.
 
 <!-- prettier-ignore -->
 [http]: https://github.com/googleapis/googleapis/blob/master/google/api/http.proto
+[routing]: https://github.com/googleapis/googleapis/blob/master/google/api/routing.proto
 [rfc 6570 §3.2.2]: https://tools.ietf.org/html/rfc6570#section-3.2.2
 
 ## Changelog
```

Comment:
Oh don't forget to update the Changelog pre-merge.

Reply:
:+1: